pandas/core/indexes/base.py

```diff
@@ -2256,15 +2256,15 @@ def intersection(self, other):
             indexer = indexer.take((indexer != -1).nonzero()[0])
         except:
             # duplicates
-            indexer = Index(other._values).get_indexer_non_unique(
-                self._values)[0].unique()
+            indexer = algos.unique1d(Index(other._values).get_indexer_non_unique(
```

Comment:
Also would have to do numpy.unique(Index(other._values).get_indexer_non_unique(self._values)[0]) as get_indexer_non_inque returns np.ndarray which does not have a unique method.